### PR TITLE
add missing @Override annotations

### PR DESCRIPTION
**Pull request by csimons**
Adding these annotations increases safety through increased compile-time checking and allows for better static analysis and tooling.

### CODE REVIEW

**Comment by KostyaSha** [comment]
Be careful, such changes may make holy war. I used your PRs as example of problems for new-comers https://groups.google.com/forum/#!topic/jenkinsci-dev/8fjvXGYbFJ4 feel free to join discussion.

**Comment by csimons** [comment]
I disagree that the presence of these annotations is a matter of style; they perform a function in conjunction with the compiler.

In addition to the increase compiler checks, the specific methods I annotated with this set of changes are overriding methods that also use the @inheritDoc annotation; if their signature is changed in a manner that violates the @Override checks, the Javadoc system will cease to generate the appropriate documentation for them.  Thus, including these means that if this happens we will get a notification in the form of a compiler error.

File: core/src/main/java/hudson/model/AbstractBuild.java [hunk 1]
```diff
@@ -987,7 +987,8 @@ public Calendar due() {
      * {@inheritDoc}
      * The action may have a {@code summary.jelly} view containing a {@code <t:summary>} or other {@code <tr>}.
      */
-    @Override public void addAction(Action a) {
+    @Override
+    public void addAction(Action a) {
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/AbstractBuild.java]: I mean such changes, this is special @jglick coding style.

File: core/src/main/java/hudson/model/LoadStatistics.java [hunk 2]
```diff
@@ -560,7 +562,9 @@ public boolean equals(Object o) {
             return true;
         }
 
-        /** {@inheritDoc} */
+        /**
+         * {@inheritDoc}
```
Comment (KostyaSha) [inline core/src/main/java/hudson/model/LoadStatistics.java]: And this changes are @stephenc personal styling AFAIK

File: core/src/main/java/hudson/model/AbstractBuild.java (hunk 1 above)
Comment (csimons) [inline core/src/main/java/hudson/model/AbstractBuild.java]: Ah, I forgot about the ancillary formatting changes in the PR.  I will back these out as they are nonessential.

Comment (KostyaSha) [inline core/src/main/java/hudson/model/AbstractBuild.java]: Better wait for other people feedback.

**Comment by batmat** [comment]
LGTM.  :+1: 
Indeed I also agree that @Override annotations are really a must-have in a Java 5+ codebase. Thanks!

**Comment by stephenc** [comment]
:+1: (and FTR my opinion is that squashing commits is evil, so I request no squashing)

**Comment by batmat** [comment]
Well @stephenc the author already rewrote (and that's a Good Thing :copyright: IMO) to remove unwanted modif to stay. But in the current state, I thoroughly agree with you. Commits are cleanly separated and that would be a crime to squash them :).

**Comment by KostyaSha** [comment]
@stephenc i will repeat that nobody requested squashing for such PRs. This commits has logical separation for better cherry-picking. Squashing topic was proposed only for PRs full of 'coding style cleanup' commits where different files covered in one commit with useless changes.

**Comment by stephenc** [comment]
I am not against asking the author to reorg their commits, I am against anyone other than the original author modifying their commits in merging the PR. If the merger want to make changes to the code, do those in follow up commits, but do not modify the original authors commits.

FTR these commits are cleanly organized, which is why I am against squashing them

**Comment by oleg-nenashev** [comment]
:+1:

**Pull request merged**